src/Handler/ContainerHandler.php

```diff
@@ -97,17 +98,20 @@ public static function afterMethodCallAnalysis(
                 );
             }
 
-            if ($service->isPublic()) {
-                $class = $service->getClassName();
-                if ($class) {
-                    $codebase->classlikes->addFullyQualifiedClassName($class);
-                    $return_type_candidate = new Union([new TNamedObject($class)]);
+            $class = $service->getClassName();
+            if ($class) {
+                $codebase->classlikes->addFullyQualifiedClassName($class);
+                $return_type_candidate = new Union([new TNamedObject($class)]);
+            }
+
+            if (!$service->isPublic()) {
+                $isTestContainer = $context->parent && (KernelTestCase::class === $context->parent || is_subclass_of($context->parent, KernelTestCase::class));
```

Comment:
hey, this isn't working for me because `$context->parent` is null. any idea why that might occur?

Reply:
can't this be simplified to
```php
$isTestContainer = is_subclass_of($context->self, KernelTestCase::class);
```
?

it works for me then

Reply:
another suggestion from @muglug 
```php
$isTestContainer = is_subclass_of($statements_source->getParentFQCLN(), KernelTestCase::class);
```

Reply:
@bendavies hey. Can you please share class declaration part of your test class?

`class TestService extends KernelTestCase`, `class TestService extends WebTestCase`

https://github.com/psalm/psalm-plugin-symfony/blob/master/tests/acceptance/acceptance/TestContainerService.feature

Reply:
```php
class ReportAccountRegisterTest extends InternalTestCase extends InternalTestCase2 extends InternalTestCase23 extends Symfony\Bundle\FrameworkBundle\Test\WebTestCase extends Symfony\Bundle\FrameworkBundle\Test\KernelTestCase
```